### PR TITLE
#162156678 Fix(validate role field): Role field should display an error when empty

### PR DESCRIPTION
**Pull request by mikenthiwa**
#### What does this PR do?
 * Validate roles field in profile and request page.
 

#### Description of Task to be completed?
* Role field should display an error when the field is cleared.

#### How should this be manually tested?
 * Clone this repo: `https://github.com/andela/travel_tool_front.git`.
 * Checkout to: `bg-validate-role-field-#162156678`.
 * Install dependencies: `yarn install`.
 * Run app: `yarn start`.

Test roles field in new requets modal
 * Click on `My Request`.
 * Click on `New Request`.
 * Clear `Your Roles` fileds.  

Test roles field in profile page
 * Click on `Profile`.
 * In personal details form clear `name` field.
 

#### Screenshots (if appropriate)
screen shot 2018-11-27 at 19 26 35

screen shot 2018-11-27 at 19 30 26

#### What are the relevant pivotal tracker stories?
[162156678](https://www.pivotaltracker.com/story/show/162156678)
#### Questions:

### CODE REVIEW

**Comment by janetnim** [comment]
Are you also validating whether input given is not valid? Like if I edit to a role that does not exist.

**Comment by mikenthiwa** [comment]
@janetnim That is out of my scope.

**Comment by amarachukwu-agbo** [comment]
@mikenthiwa , please fix your PR Title and the PR badge

**Comment by mikenthiwa** [comment]
@amarachukwu-agbo Done

File: src/components/Forms/FormsAPI/Input/index.jsx [hunk 1]
```diff
@@ -40,12 +40,18 @@ class Input extends PureComponent {
...
+        onChange: onChange || eventHandlers.handleSelectDropdown,
+      };
+      return DropdownSelect;
     case 'filter-dropdown-select':
       this.props = {
         ...this.props,
         onChange: onChange || eventHandlers.handleSelectDropdown,
+        onBlur: eventHandlers.handleInputBlur
       };
-      return this.switchDropdownInputTypes(type);
```
Comment (amarachukwu-agbo) [inline src/components/Forms/FormsAPI/Input/index.jsx]: Is there a reason why this was removed and the code above refactored? If not, then we can leave it as it was before as it fixes the codeclimate issue with this PR

Comment (mikenthiwa) [reply src/components/Forms/FormsAPI/Input/index.jsx]: @amarachukwu-agbo There is no need since it works fine with `this.switchDropdownInputTypes(type)`

Comment (amarachukwu-agbo) [reply src/components/Forms/FormsAPI/Input/index.jsx]: @mikenthiwa, please check develop on github for that file and use it to revert the changes

**Comment by amarachukwu-agbo** [comment]
LGTM

**Pull request merged**